api/all.yaml

```diff
@@ -1490,6 +1832,206 @@ components:
+          type: integer
+          format: int64
+        model_id:
+          description: Unique identifier for the model.
+          type: string
+    ModelVersionInfo:
+      type: object
+      properties:
+        model_name:
+          description: Name of model, relative to parent schema.
+          type: string
+        catalog_name:
+          description: Name of parent catalog.
+          type: string
+        schema_name:
+          description: Name of parent schema relative to its parent catalog.
+          type: string
+        version:
+          description: The version number of the model version.
+          type: integer
+          format: int64
+        source:
+          description: URI indicating the location of the source model artifacts.
+          type: string
+        run_id:
+          description: The run id used by the ML package that generated the model.
+          type: string
+        status:
+          $ref: '#/components/schemas/ModelVersionStatus'
+        storage_location:
+          description: Storage root URL for model version (for **MANAGED**, **EXTERNAL** model versions)
+          type: string
+        comment:
+          description: User-provided free-form text description.
+          type: string
+        created_at:
+          description: Time at which this model version was created, in epoch milliseconds.
+          type: integer
+          format: int64
+        updated_at:
+          description: Time at which this model version was last modified, in epoch milliseconds.
+          type: integer
+          format: int64
+        model_version_id:
+          description: Unique identifier for the model version.
+          type: string
+    ModelVersionStatus:
+      description: |
+        All possible model version statuses (MODEL_VERSION_STATUS_UNKNOWN, PENDING_REGISTRATION, FAILED_REGISTRATION, READY).
+        PENDING_REGISTRATION status indicates that the client has not completely written all model artifacts to external storage.
+        FAILED_REGISTRATION status indicates that the client has failed to write all model artifacts to external storage.
+        READY status indicates that the client has successfully written all model artifacts to external storage.
+      type: string
+      enum:
+        - MODEL_VERSION_STATUS_UNKNOWN
+        - PENDING_REGISTRATION
+        - FAILED_REGISTRATION
+        - READY
+    CreateRegisteredModel:
+      type: object
+      properties:
+        model_name:
+          description: Name of model, relative to parent schema.
+          type: string
+        catalog_name:
+          description: Name of parent catalog.
+          type: string
+        schema_name:
+          description: Name of parent schema relative to its parent catalog.
+          type: string
+        storage_location:
+          description: Storage root URL for registered model (for **MANAGED**, **EXTERNAL** regisetered models)
```

Comment:
Do we have managed/external concept for model version in UC OSS? What's the difference between them?

Reply:
We would have to.  Model versions are comprised of artifact files that make up a model version.  Those files need to be stored in a location (which is why we need a scoped token via the credential API) that we can read/write those models to/from.